File: src/main/java/com/polygloat/dtos/response/RepositoryDTO.java
package com.polygloat.dtos.response;

import com.polygloat.exceptions.InvalidStateException;
import com.polygloat.model.Permission;
import com.polygloat.model.Repository;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RepositoryDTO {
    Long id;

    String name;

    Permission.RepositoryPermissionType permissionType;

    public static RepositoryDTO fromEntityAndPermission(Repository repository, Permission permission) {
        return new RepositoryDTO(repository.getId(), repository.getName(), permission.getType());
    }
}

File: src/main/kotlin/io/polygloat/dtos/response/RepositoryDTO.kt
package io.polygloat.dtos.response

import io.polygloat.model.Permission
import io.polygloat.model.Permission.RepositoryPermissionType
import io.polygloat.model.Repository

data class RepositoryDTO(var id: Long? = null,
                         var name: String? = null,
                         var permissionType: RepositoryPermissionType? = null) {

    companion object {
        @JvmStatic
        fun fromEntityAndPermission(repository: Repository, permission: Permission): RepositoryDTO {
            return RepositoryDTO(repository.id, repository.name, permission.type)
        }
    }
}